protected/controllers/GpController.php

```diff
@@ -108,14 +108,21 @@ public function performGpSave(Contact $contact, Gp $gp, $isAjax = false)
                         $this->redirect(array('view', 'id' => $gp->id));
                     }
                 } else {
+                    if ($isAjax) {
+                        throw new CHttpException(400,"Unable to save Gp contact");
+                    }
                     $transaction->rollback();
                 }
             } else {
+                if ($isAjax) {
+                    throw new CHttpException(400,"Unable to save Gp contact");
+                }
                 $transaction->rollback();
             }
         } catch (Exception $ex) {
             OELog::logException($ex);
             $transaction->rollback();
+            throw new CHttpException(400,"Unable to save Gp contact");
```

Comment:
We may need to wrap this in an if statement based on the value of isAjax, as I suspect this may also capture other errors otherwise and just raise them as error 400s when they could end up being other exceptions entirely.